### DISCUSSION TITLE
Question for Thijs - Service Calls - Grid Export Limits and Sensors?

### DISCUSSION THREAD

**Discussion by Roving-Ronin** [wlcrs/huawei_solar#719] [Setting up the integration]
Hi Thijs,

Could you please assist, trying to work out when a user calls the following services:
- huawei_solar.set_maximum_feed_grid_power
- huawei_solar.set_maximum_feed_grid_power_percent
- huawei_solar.reset_maximum_feed_grid_power

Is there a sensor that this value is stored in that can be called/viewed with a 'states('sensor.?????')' query ?
Trying to follow the flow in the library and integration but can't find any actual sensors that report the current Grid Export Limit for the inverters. Thought it was the 'number.inverter_1_power_derating' and 'number.inverter_2_power_derating' but this just appears to be a capping on the actual inverters PV conversion to Active Power, but an actual limit upon the amount it/they can export to the Grid.

After calling these services on my own inverters, I can't see anything change in the HA sensors, nor on the sDongle that appears to usually be responsible for the ability to control Grid Exports (especially when using multiple inverters). As per below off FusionSolar portal view:

![image](https://github.com/wlcrs/huawei_solar/assets/108674933/32da5e3a-42b4-4b29-822f-0092d9aa6797)

Thanks for any assist you can give !

**Comment by Roving-Ronin** [comment]
@wlcrs apologies for tagging you, but would you please be able to assist with this query?  Many thanks !

**Comment by wlcrs** [comment]
No, these values are not exported as a sensor into HA. They can only be set via the services.

**Reply by Roving-Ronin** [reply]
Hi Thijs, 
Assuming the services though are writing to some other sensors though, is there no way to do a read of them and so be able to add present in the HA GUI showing the current export watts allowed (via a custom sensor in yaml) ? Thanks !
